lib/common/xml.c

```diff
@@ -1359,6 +1360,13 @@ crm_xml_escape(const char *text)
      * when necessary.
      */
 
+    if (text == NULL) {
+        return NULL;
+    }
+
+    length = 1 + strlen(text);
+    copy = strdup(text);
+    CRM_ASSERT(copy != NULL);
     for (size_t index = 0; index < length; index++) {
```

Comment:
If text is "" instead of NULL, length will be 1 so this loop will happen once, even though it may nto actually do anything.  We could prevent the loop from ever running with:
```
if (pcmk__str_empty(text) {
    return text;
}
```

Reply:
We still need to strdup() it and return the copy, and one run through the loop will just check a switch that hits the first case, so I think it's fine to leave.

It might be a better design to have it return NULL if the text is unchanged, to avoid the memory duplication if possible, but it's public API so we couldn't do that now without some extra hassle.